hw_diag/utilities/system_metrics.py

```diff
@@ -0,0 +1,89 @@
+from typing import Dict, Any, Union
+import logging
+import os
+import json
+from balena_supervisor import BalenaSupervisor
+
+log = logging.getLogger()
+
+
+def get_balena_metrics() -> Dict:
+    try:
+        supervisor = BalenaSupervisor.new_from_env()
+        app_state = json.loads(supervisor.get_application_state())
+    except Exception as e:
+        log.error(f"error while getting container state from supervisor {e}")
+        return {"status": "error"}
+
+    if app_state["status"] != "success":
+        log.error("container is reporting ")
+        return {"status": "error"}
+
+    metrics = {}
+    metrics["status"] = app_state["status"]
+    if (metrics["status"] == "success"):
+        metrics["containers"] = app_state["containers"]
+        metrics["release"] = app_state["release"]
+    return metrics
+
+
+def read_proc_file(file_path: str, default_value: Union[str, int]) -> Any:
+    try:
+        with open(file_path, 'r') as f:
+            return f.read()
+    except Exception as e:
+        log.warning(f"can't read proc file {file_path} {e}")
+        return default_value
+
+
+def get_serial_number() -> str:
+    try:
+        with open("/proc/device-tree/serial-number", 'r') as f:
+            return f.readline().rstrip('\x00')
+    except Exception as e:
+        log.warning(f"can't read serial number {e}")
+        return ""
+
+
+def get_variant() -> str:
+    return os.getenv('VARIANT', "")
+
+
+def get_region_override() -> str:
+    return os.getenv('REGION_OVERRIDE', "")
+
+
+def get_firmware_version() -> str:
+    return os.getenv('FIRMWARE_VERSION', "")
+
+
+def get_network_statistics(interface_list=['wlan0', 'eth0']) -> Dict:
```

Comment:
tried psutil package. But it looks like psutil doens't work very well inside container due to changes in /proc filesystem.